### ISSUE TITLE
Setup blocks seem to run on every test

### ISSUE THREAD

**Issue by nalanj** [itstommymorgan/nutrasuite#2]
I had a class with two `a` blocks in it, each with setups, and when I added the second one things started failing. I put some debug output into one of the setup blocks and realized that it was being called after each test was run.

**Comment by itstommymorgan** [comment]
So, it turns out that this is a strange ActiveSupport issue - we can't reliably override ActiveSupport::TestCase's own definition of setup for some reason. Rather than try to dig into that, I figure that Nutrasuite is young enough that we can just change the syntax; I just pushed a new version (0.2.2) that declares "before" and "after" instead of "setup" and "teardown" - that solves your problem, I think.

I've marked "setup" and "teardown" as deprecated and will remove them for 0.3.0.

**Comment by nalanj** [comment]
:+1: Thanks for figuring that out. I'll update my test accordingly.

**Comment by itstommymorgan** [comment]
Re-opening this issue: according to @gma, before blocks still don't work quite right inside ActiveSupport... so we may need another solution here.

**Comment by gma** [comment]
Let me know if you can't reproduce it. I wrote a test to exercise it without ActiveSupport present and it behaved properly, which is why I concluded ActiveSupport had probably broken it. I haven't proved that though. It was with nutrasuite 0.2.2.

**Comment by itstommymorgan** [comment]
@gma I don't doubt that's the issue, as you can see above we had a similar (but slightly different) situation with `setup` above. I'm trying to decide if there's some way we can work around ActiveSupport's use of those methods, or if we just have to pick totally new names for those methods in nutrasuite.

**Comment by itstommymorgan** [comment]
@gma actually, can you drop in a quick pseudo-code example of the problem you're seeing? I just want to be sure I'm looking at the right problem.

I'm adding a test file in specifically for ActiveSupport, so we can just go ahead and get this nonsense knocked out.

**Comment by gma** [comment]
Sure. As soon as it broke the first thing I did was write a nutrasuite test, intending to send a pull request in showing it broken. It passes obviously, because whatever is breaking it isn't loaded. It's here: gma/nutrasuite@c34da681c776f49badc4032d3410cc716a3a902f.

It could be ActiveSupport, but I notice that minitest-rails pulls in `minitest/spec` (which has `before` and `after` blocks).

**Comment by itstommymorgan** [comment]
I'm curious, what are you using minitest-rails for?

**Comment by gma** [comment]
Simply because `minitest/spec` looked like the best way to get shoulda style context blocks in minitest, with Rails 3.2. Given that nested describe blocks don't work with TConsole, and it's oriented towards people who want to do rspec style assertions, I'm guessing I could drop it entirely. I'll try that and see if the `before` issue goes away. Hang on…

**Comment by gma** [comment]
Removing `minitest/spec` does appear to sort things out. How did the minitest landscape get to be such a mess so quickly? Only in Ruby!

**Comment by itstommymorgan** [comment]
minitest/spec seems to be a problem in general, which is one of the reasons we put nutrasuite together.

I think we'd probably be ok making the two mutually exclusive, since nutrasuite explicitly tries to cover a lot of the same functionality. Thoughts?

**Comment by gma** [comment]
Trying to actively use them both in the same project would be a bit weird.

I ended up trying nutrasuite because minitest/spec doesn't work with TConsole (tests in nested `describe` blocks are hidden), and I really want to get TConsole going. I plan on writing a blog post about all the different things I've tried over the last week to get my tests running painlessly (it's probably a series of blog posts), and from what I've seen so far TConsole with nutrasuite is the best option by some margin.
